src/neml2/solvers/NewtonWithLineSearch.cxx

```diff
@@ -104,6 +124,16 @@ NewtonWithLineSearch::linesearch(NonlinearSystem & system,
                            alpha.batch_index({at::logical_not(stop)}) / _linesearch_sigma);
   }
 
+  if (at::max(crit).item<Real>() < 0)
+    neg_crit = true;
+
+  if (neg_crit && _check_crit)
+    std::cerr << "WARNING: Line Search produces negative stopping "
+                 "criteria, this could lead to convergence issue. Try with other "
+                 "linesearch_type, increase linesearch_cutback "
+                 "or reduce linesearch_stopping_criteria"
+              << std::endl;
```

Comment:
```suggestion
  if (_check_crit)
    if (at::max(crit).item<Real>() < 0)
      std::cerr << "WARNING: Line Search produces negative stopping "
                   "criteria, this could lead to convergence issue. Try with other "
                   "linesearch_type, increase linesearch_cutback "
                   "or reduce linesearch_stopping_criteria"
                << std::endl;
```